security/integration/main.go

```diff
@@ -225,70 +293,64 @@ func runCAForNodeAgentTests() error {
 	if err != nil {
 		return fmt.Errorf("failed to check certificate update node-agent (err: %v)", err)
-	} else {
-		glog.Info("Certificate of NodeAgent was updated and verified successfully")
-	}
-
-	// Delete created services
-	err = utils.DeleteService(opts.clientset, opts.namespace, ca_service.GetName())
-	if err != nil {
-		return fmt.Errorf("failed to delete CA service: %v (error: %v)", ca_service.GetName(), err)
-	}
-
-	err = utils.DeleteService(opts.clientset, opts.namespace, na_service.GetName())
-	if err != nil {
-		return fmt.Errorf("failed to delete NodeAgent service: %v (error: %v)", na_service.GetName(), err)
 	}
 
-	// Delete created pods
-	err = utils.DeletePod(opts.clientset, opts.namespace, ca_pod.GetName())
-	if err != nil {
-		return fmt.Errorf("failed to delete CA pod: %v (error: %v)", ca_pod.GetName(), err)
-	}
-
-	err = utils.DeletePod(opts.clientset, opts.namespace, na_pod.GetName())
-	if err != nil {
-		return fmt.Errorf("failed to delete NodeAgent pod: %v (error: %v)", na_pod.GetName(), err)
-	}
+	glog.Info("Certificate of NodeAgent was updated and verified successfully")
 
 	return nil
 }
 
+func cleanUpCAForNodeAgentTests() {
+	glog.Info("Removing services: istio-ca, node-agent")
+	_ = utils.DeletePod(opts.clientset, opts.namespace, "istio-ca")
+	_ = utils.DeletePod(opts.clientset, opts.namespace, "node-agent")
+	glog.Info("Removing pods: istio-ca-cert, node-agent-cert")
+	_ = utils.DeletePod(opts.clientset, opts.namespace, "istio-ca-cert")
+	_ = utils.DeletePod(opts.clientset, opts.namespace, "node-agent-cert")
+	glog.Info("Removing secrets: istio.default")
+	_ = utils.DeleteSecrets(opts.clientset, opts.namespace, "istio.default")
+}
+
 func cleanUpIntegrationTest(cmd *cobra.Command, args []string) error {
 	return utils.DeleteTestNamespace(opts.clientset, opts.namespace)
 }
 
 // This method examines the content of an Istio secret to make sure that
 // * Secret type is correctly set;
 // * Key, certificate and CA root are correctly saved in the data section;
-func examineSecret(secret *v1.Secret, expectedID string) error {
+// Returns []byte type of key, cert, root and error
+func examineSecret(secret *v1.Secret) ([]byte, []byte, []byte, error) {
 	if secret.Type != controller.IstioSecretType {
-		return fmt.Errorf(`Unexpected value for the "type" annotation: expecting %v but got %v`,
+		return nil, nil, nil, fmt.Errorf(`unexpected value for the "type" annotation: expecting %v but got %v`,
 			controller.IstioSecretType, secret.Type)
 	}
 
 	for _, key := range []string{controller.CertChainID, controller.RootCertID, controller.PrivateKeyID} {
 		if _, exists := secret.Data[key]; !exists {
-			return fmt.Errorf("%v does not exist in the data section", key)
+			return nil, nil, nil, fmt.Errorf("%v does not exist in the data section", key)
 		}
 	}
 
 	key := secret.Data[controller.PrivateKeyID]
 	cert := secret.Data[controller.CertChainID]
 	root := secret.Data[controller.RootCertID]
+
+	// should be valid
```

Comment:
Confused by the comment: what should be valid?

Reply:
Removed